core/src/main/scala/org/bitcoins/core/protocol/ln/LnHumanReadablePart.scala

```diff
@@ -3,13 +3,13 @@ package org.bitcoins.core.protocol.ln
 import org.bitcoins.core.config.NetworkParameters
 import org.bitcoins.core.protocol.ln.LnParams._
 import org.bitcoins.core.protocol.ln.currency.{LnCurrencyUnit, LnCurrencyUnits}
-import org.bitcoins.core.util.Bech32
+import org.bitcoins.core.util.{Bech32, Bech32HumanReadablePart}
 import scodec.bits.ByteVector
 
 import scala.util.matching.Regex
 import scala.util.{Failure, Success, Try}
 
-sealed abstract class LnHumanReadablePart {
+sealed abstract class LnHumanReadablePart extends Bech32HumanReadablePart {
```

Comment:
I think i would rather move all of these `HumanReadablePart`s to one file and create a proper ADT

Reply:
I'm not completely sure what you mean here. Do you want `HumanReadablePart`, `LnHumanReadablePart` and `BtcHumanReadablePart` to live in one file? And `LnHumanReadablePart` is an ADT right now?